Update `modules.md`: `pyproject.toml`, `"shared_key": "read_count"`, ruffify (#2371)
MultiQC/MultiQC: docs/core/development/modules.md

`__init__.py` file with:

```python
from .modname import MultiqcModule

__all__ = ["MultiqcModule"]
```

### Entry points

Once your submodule files are in place, you need to tell MultiQC that they
are available as an analysis module. This is done within `pyproject.toml` using
[entry points](https://setuptools.pypa.io/en/latest/userguide/entry_point.html).
In `pyproject.toml` you will see some code that looks like this:

> ```python

Reviewer: ````suggestion
```toml
````

And for the read count of long reads: `long_read_count_multiplier`, `long_read_count_prefix` and
`long_read_count_desc`.

Note that adding e.g. `"shared_key": "read_count"` will automatically add corresponding

> `description`, `modify`, and `suffix` into the column, so in usual cases this specifications
> would be sufficient:

Reviewer: ```suggestion
`description`, `modify`, and `suffix` into the column, so in most cases the following
will be sufficient:
```
